weldx/core.py

```diff
@@ -615,10 +608,21 @@ def interp_time(
                 f"{self._interp_counter} time(s)."
             )
 
+        # prepare timedelta values for internal interpolation
+        time = Time(time)
+        if time.is_absolute:
+            if self._reference_time is not None:
+                time_interp = time - self._reference_time
+            else:
+                time_interp = time - time.reference_time
+        else:
+            time_interp = time
+        time_interp = time_interp.as_pandas_index()
+
         if isinstance(self._data, xr.DataArray):
-            dax = self._interp_time_discrete(time)
+            dax = self._interp_time_discrete(time_interp)
         else:
-            dax = self._interp_time_expression(time, time_unit)
+            dax = self._interp_time_expression(time_interp, time_unit)
```

Comment:
If you don't mind, I will try to further simplify this by passing the `Time` instance directly to the internal interpolation methods.

Reply:
It was like that before (passing Time) but really didn't help because you just end up doing the conversion in both internal functions

Reply:
Just have a look, at commit 9e95f7497948a2600ec6398e3e8c309ab6d44b5d 
Maybe my comment was lacking some information to understand what I had in mind